app/models.py

```diff
@@ -0,0 +1,80 @@
+from . import db
+from datetime import datetime
+from werkzeug.security import generate_password_hash, check_password_hash
+from itsdangerous import (TimedJSONWebSignatureSerializer as Serializer,
+                          SignatureExpired, BadSignature)
+from flask import current_app
+
+
+class User(db.Model):
+    __tablename__ = 'users'
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(255), unique=True, nullable=False)
+    email = db.Column(db.String(255), nullable=False)
+    pass_hash = db.Column(db.String(255), nullable=False)
+    bucketlist = db.relationship('Bucketlist', backref='user', lazy='dynamic',
+                                 cascade='all, delete-orphan')
+
+    @property
+    def password(self):
+        raise AttributeError('password is not a readable attribute')
+
+    @password.setter
+    def password(self, password):
+        self.pass_hash = generate_password_hash(password)
+
+    def verify_password(self, password):
+        return check_password_hash(self.pass_hash, password)
+
+    def generate_confirmation_token(self, expiration=3600):
+        s = Serializer(current_app.config['SECRET_KEY'], expiration)
+        return s.dumps({'id': self.id})
+
+    @staticmethod
+    def token_confirm(token):
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except SignatureExpired:
+            """ valid token but expired """
+            return False
+        except BadSignature:
+            """ invalid token """
+            return False
+        user_id = data['id']
+        return user_id
```

Comment:
* Avoid writing unnecessary code also only create the variables that you need.
you can replace the two exceptions that you are checking with just `except:`

Consider writing line of code 33-45 like as 
```
@staticmethod
     def token_confirm(token):
        s = Serializer(current_app.config['SECRET_KEY'])
        try:
              data = s.loads(token)
        except :
              """ invalid token """
              return False
        return data['id']
```
Also variable `user_id` isn't necessary you can just use `data['id']` coz its easier to read and interprate your code.